tsconfig.json

```diff
@@ -4,9 +4,12 @@
     "baseUrl": "./src",
     "declaration": true,
     "esModuleInterop": true,
-    "lib": ["dom", "es2020"],
+    "lib": ["DOM", "ES2022"],
+    "module": "NodeNext",
     "noImplicitAny": true,
-    "outDir": "dist/"
+    "outDir": "dist/",
+    "strict": true,
+    "target": "ES2022"
```

Comment:
imo this should be something more universal because otherwise the dist of this package won't be particularly portable. I think the [default](https://www.typescriptlang.org/tsconfig#target) is ES3, but we'd likely be safe with ES6/ES2015.

Reply:
ah yeah good catch! this has been a force of habit since we've been targeting Node 18 and higher but since this library runs in browsers I agree that it should be lower 📉